In [1]:
import pandas as pd
import math

In [2]:
dataset = pd.read_csv('final_datasets/all_health_appended.csv')
dataset.head()

measure_id measure_name     country  cause_id  \
0           1       Deaths    Honduras       703   
1           1       Deaths    Honduras       703   
2           1       Deaths    Honduras       703   
3           1       Deaths  Bangladesh       521   
4           1       Deaths  Bangladesh       521   

                                   cause_name  metric_id metric_name  year  \
0                    Poisoning by other means          1      Number  1990   
1                    Poisoning by other means          2     Percent  1990   
2                    Poisoning by other means          3        Rate  1990   
3  Cirrhosis and other chronic liver diseases          1      Number  1990   
4  Cirrhosis and other chronic liver diseases          2     Percent  1990   

          value         upper         lower iso3  
0     61.099487     77.418662     47.345706  HND  
1      0.002212      0.002781      0.001731  HND  
2      1.297629      1.644216      1.005527  HND  
3  26628.429130  32225.285311  20798.897909  BGD  
4      0.023659      0.028261      0.018677  BGD

In [3]:
dataset.shape

(2386752, 12)

In [4]:
dataset['measure_name'].unique()

array(['Deaths', 'YLLs (Years of Life Lost)',
       'DALYs (Disability-Adjusted Life Years)', 'Incidence',
       'YLDs (Years Lived with Disability)', 'Prevalence'], dtype=object)

In [5]:
filters = dataset[['cause_id', 'cause_name']]
filters = filters.drop_duplicates()
filters.head()

cause_id                                  cause_name
0        703                    Poisoning by other means
3        521  Cirrhosis and other chronic liver diseases
6        526                          Digestive diseases
9        527                        Peptic ulcer disease
12       528                    Gastritis and duodenitis

In [6]:
filters.reset_index(drop=True, inplace=True)
filters.rename({'cause_id': 'value', 'cause_name': 'label'}, axis=1, inplace=True)

filters.shape

(98, 2)

### Add health codes, parent, and level

In [7]:
dictionary = pd.read_csv('data/health_codes.csv')
dictionary.head()

Cause ID                                         Cause Name  Parent ID  \
0       294                                         All causes        294   
1       295  Communicable, maternal, neonatal, and nutritio...        294   
2       955       HIV/AIDS and sexually transmitted infections        295   
3       298                                           HIV/AIDS        955   
4       948           HIV/AIDS - Drug-susceptible Tuberculosis        298   

                                         Parent Name  Level  
0                                         All causes      0  
1                                         All causes      1  
2  Communicable, maternal, neonatal, and nutritio...      2  
3       HIV/AIDS and sexually transmitted infections      3  
4                                           HIV/AIDS      4

In [8]:
dictionary.rename({'Cause ID': 'value', 'Cause Name': 'label', 'Parent ID': 'parent_id', 
                   'Parent Name': 'parent_name', 'Level': 'level'}, axis=1, inplace=True)
dictionary.head()

value                                              label  parent_id  \
0    294                                         All causes        294   
1    295  Communicable, maternal, neonatal, and nutritio...        294   
2    955       HIV/AIDS and sexually transmitted infections        295   
3    298                                           HIV/AIDS        955   
4    948           HIV/AIDS - Drug-susceptible Tuberculosis        298   

                                         parent_name  level  
0                                         All causes      0  
1                                         All causes      1  
2  Communicable, maternal, neonatal, and nutritio...      2  
3       HIV/AIDS and sexually transmitted infections      3  
4                                           HIV/AIDS      4

In [9]:
# USING NAMES 
# # get unique conditions
# unique_in_filters = filters['label'].unique()
# print('\nUnique unique_in_filters:', len(unique_in_filters))
# unique_in_dictionary = dictionary['label'].unique()
# print('Unique unique_in_dictionary:', len(unique_in_dictionary))

# # find the foods not in filters
# difference_1 = list(set(unique_in_filters) - set(unique_in_dictionary))
# print('\nfoods not in filters', len(difference_1), ':', difference_1)

# # find the foods not in dictionary
# difference_2 = list(set(unique_in_dictionary) - set(unique_in_filters))
# difference_2.sort()
# print('\nfoods not in dictionary', len(difference_2), ':', difference_2)


# USING CODES 
# get unique conditions
unique_in_filters = filters['value'].unique()
print('\nUnique unique_in_filters:', len(unique_in_filters))
unique_in_dictionary = dictionary['value'].unique()
print('Unique unique_in_dictionary:', len(unique_in_dictionary))

# find the diseases not in filters
difference_1 = list(set(unique_in_filters) - set(unique_in_dictionary))
print('\ndiseases not in filters', len(difference_1), ':', difference_1)

# find the diseases not in dictionary
difference_2 = list(set(unique_in_dictionary) - set(unique_in_filters))
difference_2.sort()
print('\ndiseases not in dictionary', len(difference_2), ':', difference_2)


Unique unique_in_filters: 98
Unique unique_in_dictionary: 360

foods not in filters 2 : [1028, 1029]

foods not in dictionary 264 : [294, 295, 297, 298, 300, 302, 319, 320, 321, 322, 328, 329, 332, 337, 338, 339, 340, 341, 342, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 367, 368, 369, 370, 374, 375, 376, 379, 381, 382, 383, 384, 385, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 408, 409, 411, 414, 418, 419, 420, 423, 441, 444, 447, 450, 459, 462, 490, 492, 493, 495, 496, 497, 500, 501, 502, 503, 504, 509, 510, 511, 512, 513, 514, 516, 520, 522, 523, 524, 525, 541, 547, 548, 557, 560, 561, 562, 563, 564, 565, 566, 569, 573, 574, 579, 582, 585, 588, 592, 595, 596, 597, 598, 602, 604, 605, 606, 607, 608, 609, 612, 614, 615, 616, 618, 619, 639, 640, 641, 642, 643, 644, 645, 646, 647, 648, 649, 650, 651, 652, 653, 656, 657, 658, 659, 660, 662, 663, 665, 668, 669, 670, 671, 672, 674, 675, 679, 680, 681, 6

# Do I want to add some of the conditions not present in the current list?

# parent_id == NaN if the category is parent level
    use value and label values, add level = 1
    for [1028, 1029] parent_id, parent_name and level are NaNs (not available in the dictionary dataset, and are top level)
    then typecast parent_id and level as int (not allowed as NaN values cannot be converted)


In [10]:
mapping = pd.merge(filters, dictionary, on=["value", "label"], how='left')
mapping

value                                              label  parent_id  \
0     703                           Poisoning by other means      700.0   
1     521         Cirrhosis and other chronic liver diseases      526.0   
2     526                                 Digestive diseases      409.0   
3     527                               Peptic ulcer disease      992.0   
4     528                           Gastritis and duodenitis      992.0   
5     529                                       Appendicitis      526.0   
6     530         Paralytic ileus and intestinal obstruction      526.0   
7     531            Inguinal, femoral, and abdominal hernia      526.0   
8     532                         Inflammatory bowel disease      526.0   
9     533                      Vascular intestinal disorders      526.0   
10    534                   Gallbladder and biliary diseases      526.0   
11    535                                       Pancreatitis      526.0   
12    542                             Neurological disorders      409.0   
13    508                       Chronic respiratory diseases      409.0   
14   1028  Total burden related to Non-alcoholic fatty li...        NaN   
15   1029                                      Total cancers        NaN   
16    572                                   Eating disorders      558.0   
17    587                                  Diabetes mellitus      974.0   
18    589                             Chronic kidney disease      974.0   
19    591         Chronic kidney disease due to hypertension      589.0   
20    593  Chronic kidney disease due to other and unspec...      589.0   
21    724                             Interpersonal violence      717.0   
22    491                            Cardiovascular diseases      409.0   
23    494                                             Stroke      491.0   
24    498                         Hypertensive heart disease      491.0   
25    499                     Cardiomyopathy and myocarditis      491.0   
26    962                    Maternal and neonatal disorders      295.0   
27    971  Cirrhosis and other chronic liver diseases due...      521.0   
28    426                Tracheal, bronchus, and lung cancer      410.0   
29    429                                      Breast cancer      410.0   
..    ...                                                ...        ...   
68    484                                   Hodgkin lymphoma      410.0   
69    485                               Non-Hodgkin lymphoma      410.0   
70    486                                   Multiple myeloma      410.0   
71    487                                           Leukemia      410.0   
72    613           Hemoglobinopathies and hemolytic anemias      640.0   
73    453               Gallbladder and biliary tract cancer      410.0   
74    626                          Musculoskeletal disorders      409.0   
75    627                               Rheumatoid arthritis      626.0   
76    567                               Depressive disorders      558.0   
77    568                          Major depressive disorder      567.0   
78    570                                   Bipolar disorder      558.0   
79    571                                  Anxiety disorders      558.0   
80    575                          Autism spectrum disorders      558.0   
81    578           Attention-deficit/hyperactivity disorder      558.0   
82    972                                 Headache disorders      542.0   
83    977                                  Atopic dermatitis      654.0   
84    979                             Seborrhoeic dermatitis      654.0   
85    388                                  Iodine deficiency      386.0   
86    389                               Vitamin A deficiency      386.0   
87    390                            Dietary iron deficiency      386.0   
88    628                                     Osteoarthritis      626.0   
89    630                    

In [11]:
mapping.shape

(98, 5)

In [12]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', -1)

In [13]:
mapping.sort_values(by=["parent_name"], inplace=True)
mapping

value                                                              label  \
34  687    Injuries                                                            
25  499    Cardiomyopathy and myocarditis                                      
24  498    Hypertensive heart disease                                          
23  494    Stroke                                                              
41  997    Chronic kidney disease due to diabetes mellitus type 1              
42  998    Chronic kidney disease due to diabetes mellitus type 2              
20  593    Chronic kidney disease due to other and unspecified causes          
19  591    Chronic kidney disease due to hypertension                          
52  515    Asthma                                                              
27  971    Cirrhosis and other chronic liver diseases due to NAFLD             
56  386    Nutritional deficiencies                                            
26  962    Maternal and neonatal disorders                                     
77  568    Major depressive disorder                                           
84  979    Seborrhoeic dermatitis                                              
83  977    Atopic dermatitis                                                   
18  589    Chronic kidney disease                                              
17  587    Diabetes mellitus                                                   
39  976    Diabetes mellitus type 2                                            
38  975    Diabetes mellitus type 1                                            
11  535    Pancreatitis                                                        
10  534    Gallbladder and biliary diseases                                    
9   533    Vascular intestinal disorders                                       
8   532    Inflammatory bowel disease                                          
7   531    Inguinal, femoral, and abdominal hernia                             
6   530    Paralytic ileus and intestinal obstruction                          
5   529    Appendicitis                                                        
1   521    Cirrhosis and other chronic liver diseases                          
40  992    Upper digestive system diseases                                     
43  696    Unintentional injuries                                              
45  717    Self-harm and interpersonal violence                                
35  688    Transport injuries                                                  
36  366    Maternal disorders                                                  
55  380    Neonatal disorders                                                  
80  575    Autism spectrum disorders                                           
92  559    Schizophrenia                                                       
76  567    Depressive disorders                                                
16  572    Eating disorders                                                    
81  578    Attention-deficit/hyperactivity disorder                            
78  570    Bipolar disorder                                                    
79  571    Anxiety disorders                                                   
90  631    Neck pain                                                           
89  630    Low back pain                                                       
91  632    Gout                                                                
88  628    Osteoarthritis                                                      
75  627    Rheumatoid arthritis                                                
61  465    Ovarian cancer                                                      
60  456    Pancreatic cancer                                                   
68  484    Hodgkin lymphoma                                                    
64  474    Bladder cancer                                                      
69  

In [14]:
for index, row in mapping.iterrows():
#     if row['value'] == 1029:
    if math.isnan(row['parent_id']):
        mapping.at[index, 'parent_name']= row['label']
        mapping.at[index, 'parent_id']= row['value']
        mapping.at[index, 'level']= 999

In [15]:
mapping

value                                                              label  \
34  687    Injuries                                                            
25  499    Cardiomyopathy and myocarditis                                      
24  498    Hypertensive heart disease                                          
23  494    Stroke                                                              
41  997    Chronic kidney disease due to diabetes mellitus type 1              
42  998    Chronic kidney disease due to diabetes mellitus type 2              
20  593    Chronic kidney disease due to other and unspecified causes          
19  591    Chronic kidney disease due to hypertension                          
52  515    Asthma                                                              
27  971    Cirrhosis and other chronic liver diseases due to NAFLD             
56  386    Nutritional deficiencies                                            
26  962    Maternal and neonatal disorders                                     
77  568    Major depressive disorder                                           
84  979    Seborrhoeic dermatitis                                              
83  977    Atopic dermatitis                                                   
18  589    Chronic kidney disease                                              
17  587    Diabetes mellitus                                                   
39  976    Diabetes mellitus type 2                                            
38  975    Diabetes mellitus type 1                                            
11  535    Pancreatitis                                                        
10  534    Gallbladder and biliary diseases                                    
9   533    Vascular intestinal disorders                                       
8   532    Inflammatory bowel disease                                          
7   531    Inguinal, femoral, and abdominal hernia                             
6   530    Paralytic ileus and intestinal obstruction                          
5   529    Appendicitis                                                        
1   521    Cirrhosis and other chronic liver diseases                          
40  992    Upper digestive system diseases                                     
43  696    Unintentional injuries                                              
45  717    Self-harm and interpersonal violence                                
35  688    Transport injuries                                                  
36  366    Maternal disorders                                                  
55  380    Neonatal disorders                                                  
80  575    Autism spectrum disorders                                           
92  559    Schizophrenia                                                       
76  567    Depressive disorders                                                
16  572    Eating disorders                                                    
81  578    Attention-deficit/hyperactivity disorder                            
78  570    Bipolar disorder                                                    
79  571    Anxiety disorders                                                   
90  631    Neck pain                                                           
89  630    Low back pain                                                       
91  632    Gout                                                                
88  628    Osteoarthritis                                                      
75  627    Rheumatoid arthritis                                                
61  465    Ovarian cancer                                                      
60  456    Pancreatic cancer                                                   
68  484    Hodgkin lymphoma                                                    
64  474    Bladder cancer                                                      
69  

In [16]:
mapping.dtypes

value          int64  
label          object 
parent_id      float64
parent_name    object 
level          float64
dtype: object

In [17]:
mapping['parent_id'] = mapping['parent_id'].astype(int)
mapping['level'] = mapping['level'].astype(int)

In [18]:
mapping.dtypes

value          int64 
label          object
parent_id      int64 
parent_name    object
level          int64 
dtype: object

In [19]:
mapping['label'].unique()

array(['Injuries', 'Cardiomyopathy and myocarditis',
       'Hypertensive heart disease', 'Stroke',
       'Chronic kidney disease due to diabetes mellitus type 1',
       'Chronic kidney disease due to diabetes mellitus type 2',
       'Chronic kidney disease due to other and unspecified causes',
       'Chronic kidney disease due to hypertension', 'Asthma',
       'Cirrhosis and other chronic liver diseases due to NAFLD',
       'Nutritional deficiencies', 'Maternal and neonatal disorders',
       'Major depressive disorder', 'Seborrhoeic dermatitis',
       'Atopic dermatitis', 'Chronic kidney disease', 'Diabetes mellitus',
       'Diabetes mellitus type 2', 'Diabetes mellitus type 1',
       'Pancreatitis', 'Gallbladder and biliary diseases',
       'Vascular intestinal disorders', 'Inflammatory bowel disease',
       'Inguinal, femoral, and abdominal hernia',
       'Paralytic ileus and intestinal obstruction', 'Appendicitis',
       'Cirrhosis and other chronic liver diseases',
 

In [20]:
excluded_list = ['Injuries', 'Unintentional injuries', 'Self-harm and interpersonal violence', 
                 'Transport injuries', 'Poisoning by other means', 'Interpersonal violence', 
                 'Maternal and neonatal disorders', 'Maternal disorders', 'Neonatal disorders', 
                 'Low back pain', 'Chronic respiratory diseases', 'Self-harm', 
                 'Total burden related to Non-alcoholic fatty liver disease (NAFLD)', 'Total cancers']



In [21]:
curated_conditions = mapping

for condition in excluded_list:
    curated_conditions.drop(mapping[mapping['label'] == condition].index, inplace = True)
    print('Removing', condition, '...')

Removing Injuries ...
Removing Unintentional injuries ...
Removing Self-harm and interpersonal violence ...
Removing Transport injuries ...
Removing Poisoning by other means ...
Removing Interpersonal violence ...
Removing Maternal and neonatal disorders ...
Removing Maternal disorders ...
Removing Neonatal disorders ...
Removing Low back pain ...
Removing Chronic respiratory diseases ...
Removing Self-harm ...
Removing Total burden related to Non-alcoholic fatty liver disease (NAFLD) ...
Removing Total cancers ...


In [22]:
print('Injuries' in curated_conditions['label'].values)

False


In [23]:
curated_conditions

curated_conditions.sort_values(by=['label'])

value                                                       label  \
97  661    Acne vulgaris                                                
47  543    Alzheimer's disease and other dementias                      
79  571    Anxiety disorders                                            
5   529    Appendicitis                                                 
52  515    Asthma                                                       
83  977    Atopic dermatitis                                            
81  578    Attention-deficit/hyperactivity disorder                     
80  575    Autism spectrum disorders                                    
78  570    Bipolar disorder                                             
64  474    Bladder cancer                                               
65  477    Brain and central nervous system cancer                      
29  429    Breast cancer                                                
25  499    Cardiomyopathy and myocarditis                               
22  491    Cardiovascular diseases                                      
30  432    Cervical cancer                                              
18  589    Chronic kidney disease                                       
41  997    Chronic kidney disease due to diabetes mellitus type 1       
42  998    Chronic kidney disease due to diabetes mellitus type 2       
19  591    Chronic kidney disease due to hypertension                   
20  593    Chronic kidney disease due to other and unspecified causes   
1   521    Cirrhosis and other chronic liver diseases                   
27  971    Cirrhosis and other chronic liver diseases due to NAFLD      
76  567    Depressive disorders                                         
94  654    Dermatitis                                                   
37  974    Diabetes and kidney diseases                                 
17  587    Diabetes mellitus                                            
38  975    Diabetes mellitus type 1                                     
39  976    Diabetes mellitus type 2                                     
87  390    Dietary iron deficiency                                      
2   526    Digestive diseases                                           
16  572    Eating disorders                                             
10  534    Gallbladder and biliary diseases                             
73  453    Gallbladder and biliary tract cancer                         
4   528    Gastritis and duodenitis                                     
93  536    Gastroesophageal reflux disease                              
91  632    Gout                                                         
54  603    Gynecological diseases                                       
82  972    Headache disorders                                           
72  613    Hemoglobinopathies and hemolytic anemias                     
68  484    Hodgkin lymphoma                                             
24  498    Hypertensive heart disease                                   
49  545    Idiopathic epilepsy                                          
8   532    Inflammatory bowel disease                                   
7   531    Inguinal, femoral, and abdominal hernia                      
85  388    Iodine deficiency                                            
63  471    Kidney cancer                                                
71  487    Leukemia                                                     
44  417    Liver cancer                                                 
77  568    Major depressive disorder                                    
33  558    Mental disorders                                             
67  483    Mesothelioma                                                 
51  554    Motor neuron disease                                         
70  486    Multiple myeloma                                             
50  546    Multiple sclerosis                                 

In [24]:
# save all conditions to json
# filters.to_json('final_datasets/health_filters.json', orient="records")

In [25]:
# save curated conditions to json
curated_conditions.to_json('final_datasets/health_filters_curated.json', orient="records")

In [26]:
# save merged dataset 
# mapping.to_json('final_datasets/merged_health_filters.json', orient="records")